### PR TITLE
Add missing syncs for G_PM_NPRIMITIVE

### PR DESCRIPTION
**Pull request by FazanaJ**
Allows G_PM_NPRIMITIVE to function fully without crashing or visual artefacting, providing minor speedup on the RDP.

### CODE REVIEW

**Review by thecozies — APPROVED** [review]
lgtm

**Review by thecozies — CHANGES_REQUESTED** [review]
make some tweaks or something

**Comment by FazanaJ** [comment]
I have reviewed my own code and have found myself to have done no wrongdoing

**Review by thecozies — APPROVED** [review]
pretty suspicious, still looks good

File: actors/sparkle_animation/model.inc.c [hunk 1]
```diff
@@ -46,6 +46,7 @@ const Gfx sparkles_animation_seg4_dl_04035288[] = {
     gsSPVertex(sparkles_animation_seg4_vertex_04032A48, 4, 0),
     gsSP2Triangles( 0,  1,  2, 0x0,  0,  2,  3, 0x0),
     gsSPTexture(0xFFFF, 0xFFFF, 0, G_TX_RENDERTILE, G_OFF),
+    gsDPPipeSync(),
```
Comment (Mr-Wiseguy) [inline actors/sparkle_animation/model.inc.c:49]: Move this one line down

File: bin/debug_level_select.c [hunk 2]
```diff
@@ -216,6 +216,7 @@ const Gfx debug_level_select_dl_07000858[] = {
     gsSPVertex(debug_level_select_vertex_07000798, 12, 0),
     gsSP2Triangles( 0,  1,  2, 0x0,  3,  4,  5, 0x0),
     gsSP2Triangles( 6,  7,  8, 0x0,  9, 10, 11, 0x0),
+    gsDPPipeSync(),
```
Comment (Mr-Wiseguy) [inline bin/debug_level_select.c:219]: Not needed

File: bin/debug_level_select.c [hunk 3]
```diff
@@ -851,6 +854,7 @@ const Gfx debug_level_select_dl_070025F0[] = {
     gsSPVertex(debug_level_select_vertex_07002530, 12, 0),
     gsSP2Triangles( 0,  1,  2, 0x0,  3,  4,  5, 0x0),
     gsSP2Triangles( 6,  7,  8, 0x0,  9, 10, 11, 0x0),
+    gsDPPipeSync(),
```
Comment (Mr-Wiseguy) [inline bin/debug_level_select.c:857]: Not needed

File: bin/debug_level_select.c [hunk 4]
```diff
@@ -1126,6 +1130,7 @@ const Gfx debug_level_select_dl_07003258[] = {
     gsSP1Triangle(12, 13, 14, 0x0),
     gsSPVertex(debug_level_select_vertex_07003228, 3, 0),
     gsSP1Triangle( 0,  1,  2, 0x0),
+    gsDPPipeSync(),
```
Comment (Mr-Wiseguy) [inline bin/debug_level_select.c:1133]: Not needed

File: bin/debug_level_select.c [hunk 5]
```diff
@@ -1362,6 +1367,7 @@ const Gfx debug_level_select_dl_07003DB8[] = {
     gsSPVertex(debug_level_select_vertex_07003D28, 9, 0),
     gsSP2Triangles( 0,  1,  2, 0x0,  3,  4,  5, 0x0),
     gsSP1Triangle( 6,  7,  8, 0x0),
+    gsDPPipeSync(),
```
Comment (Mr-Wiseguy) [inline bin/debug_level_select.c:1370]: Not needed

**Review by Mr-Wiseguy — CHANGES_REQUESTED** [review]

File: bin/segment2.c [hunk 6]
```diff
@@ -2232,6 +2232,7 @@ const Gfx dl_draw_text_bg_box[] = {
     gsDPSetRenderMode(G_RM_XLU_SURF, G_RM_XLU_SURF2),
     gsSPVertex(vertex_text_bg_box, 4, 0),
     gsSP2Triangles( 0,  1,  2, 0x0,  0,  2,  3, 0x0),
+    gsDPPipeSync(),
```
[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply bin/segment2.c:2235]: Still here

File: actors/sparkle_animation/model.inc.c (hunk 1 above)
Comment (Mr-Wiseguy) [reply actors/sparkle_animation/model.inc.c:49]: Move

**Review by Mr-Wiseguy — CHANGES_REQUESTED** [review]

**Review by Mr-Wiseguy — APPROVED** [review]
looketh good to me

**Pull request merged**